src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -611,25 +611,52 @@ internal int GetChildIndex(int slot)
         /// efficient implementations.
         /// </summary>
         internal virtual int GetChildPosition(int index)
+        {
+            return GetChildPosition(index, useNextNotPrevious: false);
+        }
+
+        internal int GetChildPosition(int index, bool useNextNotPrevious)
         {
             int offset = 0;
             var green = this.Green;
-            while (index > 0)
+
+            if (!useNextNotPrevious)
```

Comment:
nit: flip?

Reply:
I used `!useNextNotPrevious` to minimize the differences in the PR, and also because `!useNextNotPrevious` is the common case.